client/src/components/CustomerDetails.js

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from "react";
+import { gql, useMutation, useQuery } from '@apollo/client';
+
+import { useHistory } from "react-router-dom";
+
+const CUSTOMER_DETAILS = gql`
+    mutation EditCustomer($friendlyName:String! $phoneNumber:String!) {
+        createUser(friendlyName:$friendlyName, phoneNumber:$phoneNumber) {
+            id,
+            friendlyName,
+            phoneNumber 
+        }
+    }
+`;
+
+const FETCH_CUSTOMER = gql`
+    query($id: ID!) {
+        customer(id: $id) {
+            friendlyName
+            phoneNumber
+        }
+    }
+`;
+
+const CustomerDetails = () => {
+    
+    const history = useHistory();
+    const manageCustomer = () => {
+    history.push("/manage-customer");
+    };
+    const urlInfo = window.location.href.split('/');
+    const KnownUserId = urlInfo[4];
+    
+    const [ editCustomerName, setEditCustomerName ] = useState('');
+    const [ editCustomerNumber, setCustomerNumber ] = useState('');
+
+    const FetchCustomerName = (userId) => {
+ 
+        const {loading, error, data } = useQuery(FETCH_CUSTOMER, {
+            variables: {id: userId},
+        });
+        
+        if(loading) return <p>Loading...</p>;
+        if(error) return <p>Error :(</p>;
+    
+        return (
+            data.customer.friendlyName
+        )            
+    };
+
+    const FetchCustomerNumber = (userId) => {
+ 
+        const {loading, error, data } = useQuery(FETCH_CUSTOMER, {
+            variables: {id: userId},
+        });
+        
+        if(loading) return <p>Loading...</p>;
+        if(error) return <p>Error :(</p>;
+    
+        return (
+            data.customer.phoneNumber
+        )            
+    };
+
+    const [ createUser ] = useMutation(CUSTOMER_DETAILS);
+
+    const knownUserName = FetchCustomerName(KnownUserId);
+    
+    const knownUserNumber = FetchCustomerNumber(KnownUserId);
+        
+        return(
+            <div className="container-fluid m-1">
+                <form
+                    onSubmit={e => {
+                        e.preventDefault();
+                        createUser({ variables: {friendlyName: editCustomerName, phoneNumber: editCustomerNumber }});
```

Comment:
does this call happen for existing users? If so, you'd end up with a second user with the same info.

Reply:
From my understanding, this is only for creating a new user because on the ` input`'s value field I use this `value={ editCustomerName || knownUserName }` . which means If there is a value returned from `editCustomerName`, we will use it immidiately otherwise we will take the value from `knownUserName`which got id from URL and then did a query and return a value for us. Does it make a sense?

Reply:
My understanding is that the onSubmit is triggered whenever the submit button on this page is pressed; in the case of editing the existing customer, how do you prevent onSubmit from triggering?

Reply:
This is not a problem for us now, I think you can close this issue anytime since I have already created a new branch for deleting customer.

Reply:
You can close a PR if you don't intend to work on it further.